File: dawning-admin/src/api/audit-log.ts
import axios from 'axios';
import type { IPagedData } from './paged-data';

// 审计日志接口定义
export interface AuditLog {
  id: string;
  userId?: string;
  username?: string;
  action: string;
  entityType?: string;
  entityId?: string;
  description?: string;
  ipAddress?: string;
  userAgent?: string;
  requestPath?: string;
  requestMethod?: string;
  statusCode?: number;
  oldValues?: any;
  newValues?: any;
  createdAt: string;
}

// 审计日志查询参数
export interface AuditLogQueryParams {
  page?: number;
  pageSize?: number;
  userId?: string;
  username?: string;
  action?: string;
  entityType?: string;
  entityId?: string;
  ipAddress?: string;
  startDate?: string;
  endDate?: string;
}

// 获取审计日志分页列表
export async function getAuditLogs(
  params: AuditLogQueryParams
): Promise<IPagedData<AuditLog>> {
  const response = await axios.get('/api/audit-log', { params });
  // 拦截器返回 { code, message, data }，response.data 包含 { list, pagination }
  const { list, pagination } = response.data;
  return {
    items: list,
    totalCount: pagination.total,
    pageIndex: pagination.current,
    pageSize: pagination.pageSize,
  };
}

// 根据ID获取审计日志详情
export function getAuditLogById(id: string) {
  return axios.get<AuditLog>(`/api/audit-log/${id}`);
}

// 清理指定天数之前的审计日志（仅super_admin）
export function cleanupOldAuditLogs(daysToKeep: number) {
  return axios.delete(`/api/audit-log/cleanup?daysToKeep=${daysToKeep}`);
}

File: services/Dawning.Gateway/src/Dawning.Identity.Domain/Aggregates/Administration/AuditLog.cs
using System;
using Dawning.Identity.Domain.Core.Interfaces;

namespace Dawning.Identity.Domain.Aggregates.Administration
{
    /// <summary>
    /// 审计日志聚合根
    /// </summary>
    public class AuditLog : IAggregateRoot
    {
        /// <summary>
        /// 审计日志唯一标识
        /// </summary>
        public Guid Id { get; set; }

        /// <summary>
        /// 操作用户ID
        /// </summary>
        public Guid? UserId { get; set; }

        /// <summary>
        /// 操作用户名
        /// </summary>
        public string? Username { get; set; }

        /// <summary>
        /// 操作类型 (Create, Update, Delete, Login, Logout等)
        /// </summary>
        public string Action { get; set; } = string.Empty;

        /// <summary>
        /// 实体类型 (User, Role, Application等)
        /// </summary>
        public string? EntityType { get; set; }

        /// <summary>
        /// 实体ID
        /// </summary>
        public Guid? EntityId { get; set; }

        /// <summary>
        /// 操作描述
        /// </summary>
        public string? Description { get; set; }

        /// <summary>
        /// IP地址
        /// </summary>
        public string? IpAddress { get; set; }

        /// <summary>
        /// 用户代理
        /// </summary>
        public string? UserAgent { get; set; }

        /// <summary>
        /// 请求路径
        /// </summary>
        public string? RequestPath { get; set; }

        /// <summary>
        /// 请求方法
        /// </summary>
        public string? RequestMethod { get; set; }

        /// <summary>
        /// HTTP状态码
        /// </summary>
        public int? StatusCode { get; set; }

        /// <summary>
        /// 修改前的值 (JSON字符串)
        /// </summary>
        public string? OldValues { get; set; }

        /// <summary>
        /// 修改后的值 (JSON字符串)
        /// </summary>
        public string? NewValues { get; set; }

        /// <summary>
        /// 创建时间
        /// </summary>
        public DateTime CreatedAt { get; set; }

        /// <summary>
        /// 时间戳
        /// </summary>
        public long Timestamp { get; set; }
    }
}
